code/modules/recycling/conveyor2.dm

```diff
@@ -184,13 +211,15 @@
 	speed_process = 1
 
 
-/obj/machinery/conveyor_switch/New()
-	..()
+obj/machinery/conveyor_switch/New(newloc, newid)
+	..(newloc)
+	if(!id)
+		id = newid
 	update()
 
 	spawn(5)		// allow map load
 		conveyors = list()
-		for(var/obj/machinery/conveyor/C in world)
+		for(var/obj/machinery/conveyor/C in machines)

```

Comment:
This may be why it's not working.

`world` calls are terrible, but if the `machines` list isn't populated yet when this is called, well....

Another possibility is you may need to make this `Initialize` instead of `New`